fix(app): properly deserialize server options

olevski opened the pull request:
/deploy #persist

olevski commented:
The acceptance tests are failing because PVCs cannot be mounted on dev.

Panaetius reviewed (changes requested):
`MemoryField` should probably be renamed to something more generic, like `ByteSizeField` or similar. But that's not really related to this PR.

Not sure why we need to check for `None` and empty string for storage but not the other fields

olevski commented on renku_notebooks/api/schemas/servers_get.py:
```diff
@@ -305,20 +305,24 @@ def get_resource_requests(server):
+                )
             if (
                 "gpu_request" in server_options_keys
                 and int(server_options["gpu_request"]) > 0
```
🤦 thanks Ralf I completely missed that... It worked because the gpu requests are always integers and in the worst case they are strings but with well-formed integer values... Nevertheless it was asking for trouble with that code.

olevski commented:
I addressed the comment on the GpuField. Also I renamed the MemoryField as you suggested.

I deployed and tested this on my personal deployment (which has some sessions from an older renku version) and things work.
